core/src/price_estimation.rs

```diff
@@ -156,9 +166,13 @@ impl PriceEstimating for PriceOracle {
         .boxed()
     }
 
-    fn get_eth_price(&self) -> BoxFuture<'_, Option<u128>> {
-        // TODO: Implement ETH price estimation.
-        immediate!(None)
+    fn get_eth_price(&self) -> BoxFuture<'_, Option<NonZeroU128>> {
+        async move {
+            let token_id = self.eth_token_id().await.ok()?;
```

Comment:
We need to determine the eth token id before running the solver do termine min_avg_fee, which involves fetching fetching all token infos into the cache.  Thus the previous argument that we shouldn't populate the token info cache on startup because it would introduce delay, doesn't hold anymore.
Imo we should in the constructor call cache all and also directly store the eth_token_id in PriceOracle so that we only need to call `eth_token_id` once.

Reply:
I would really like to make the cache_all use a single BatchTransport RPC call for this to be just one roundtrip instead of 80 at the moment. I started this some time last week, but got blocked by suboptimal support for it in ethcontracts (https://github.com/gnosis/ethcontract-rs/issues/355).

We could for now also hardcode the ID for WETH since it's always 1.

Reply:
It's <num of tokens> roundtrips but it does have parallelization even without batched transport.
I think I do like hardcoding it (and making the hardcoded value configurable if needed). Will change it to that.